### DISCUSSION TITLE
How to use _app.js & _document.js with Multi Zones?

### DISCUSSION THREAD

**Discussion by paulgaumer** [vercel/next.js#12295] [Help]
Hi there,

I'm building a SaaS with a dashboard side (admin) and a landing side (user). I started as one app then tried to divide both apps using multi zone. It works to some extents, but I have serious styling issues, coming from styled components I believe (works fine on local).

The docs mention that every page must be unique on each side. The thing is I have a `_app.js` needed for Context and a `_document.js` needed for styled components to work on SSR. Both cannot be renamed and therefore exist in both apps' pages folder. I feel this might be the source of my bugs.

Is there a way for me the fix this?

**Comment by Jerry2001Qu** [comment]
I'm running into the exact same issue. I don't know what to do.

**Reply by Jerry2001Qu** [reply]
My 'fix' for the time being will be launching the admin site on a completely different domain. I'll have to link between the two sites accordingly. Pretty bad solution, but I believe it'll work.

**Comment by Jerry2001Qu** [comment]
I think it might be better to use subdomains as per: https://vercel.com/docs/v2/custom-domains

Going to be trying this out.

**Comment by jnv** [comment]
Hi, I was recently digging into multi zones, although we don't use Vercel, just a regular reverse proxy. Both of our apps have custom `_app.js`, but just one of them uses styled components. Didn't run into any issues. 

I don't think custom _app or _document could be culprits, since those are never delivered as standalone pages to user. 

 Perhaps could you be more specific about the issue you are facing, preferably with minimal example for reproduction? Also, do you have `assetPrefix` setup correctly during the build?
